src/lib.rs

```diff
@@ -2583,6 +2598,12 @@ impl Bindings {
             _ => Ok(Cow::Owned(source)),
         }
     }
+
+    /// Take all the warning messages.
+    #[inline]
+    pub fn take_warnings(&mut self) -> impl Iterator<Item = String> + '_ {
```

Comment:
Can we just have `pub fn warnings(&self) -> &[String] { &self.warnings }` (effectively)?

I don't see why it's useful to take ownership of them.

Reply:
I thought it would be better to consume them so you don't have the chance to emit them twice.